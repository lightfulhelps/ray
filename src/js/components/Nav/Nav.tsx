import * as React from 'react';
import classNames from 'classnames';

type Props = {
  className?: string;
  isFill?: boolean;
  isPills?: boolean;
  isTabs?: boolean;
  withoutWrapper?: boolean;
  tag?: keyof JSX.IntrinsicElements | typeof React.Component;
  [key: string]: any;
};

export const Nav: React.FC<Props> = ({
  className,
  isFill,
  isPills,
  isTabs,
  withoutWrapper = false,
  tag: Tag = 'ul',
  ...other
}: Props) => {
  const classes = classNames(
    className,
    'nav',
    { 'nav-fill': isFill },
    { 'nav-pills': isPills && !isTabs },
    { 'nav-tabs': !isPills && isTabs }
  );

  const renderNav = () => <Tag {...other} className={classes} />;

  return withoutWrapper ? renderNav() : <div className="overflow-hidden">{renderNav()}</div>;
};

export default Nav;
