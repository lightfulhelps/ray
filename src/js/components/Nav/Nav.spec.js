import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';
import Nav from './Nav';

const setup = (overrides = {}) => {
  const props = merge({}, overrides);
  const wrapper = shallow(<Nav {...props} />);

  return { wrapper, props };
};

describe('<Nav />', () => {
  it('should render', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle className', () => {
    const { wrapper } = setup();

    wrapper.setProps({ className: 'custom' });

    expect(wrapper.hasClass('nav')).toBe(true);
    expect(wrapper.hasClass('custom')).toBe(true);
  });

  it('should handle isFill', () => {
    const { wrapper } = setup();

    expect(wrapper.hasClass('nav-fill')).toBe(false);

    wrapper.setProps({ isFill: true });

    expect(wrapper.hasClass('nav-fill')).toBe(true);
  });

  it('should handle isPills', () => {
    const { wrapper } = setup();

    expect(wrapper.hasClass('nav-pills')).toBe(false);

    wrapper.setProps({ isPills: true });

    expect(wrapper.hasClass('nav-pills')).toBe(true);
  });

  it('should handle isTabs', () => {
    const { wrapper } = setup();

    expect(wrapper.hasClass('nav-tabs')).toBe(false);

    wrapper.setProps({ isTabs: true });

    expect(wrapper.hasClass('nav-tabs')).toBe(true);
  });

  it('should handle tag', () => {
    const { wrapper } = setup();

    expect(wrapper.type()).toBe('ul');

    wrapper.setProps({ tag: 'nav' });

    expect(wrapper.type()).toBe('nav');

    wrapper.setProps({ tag: 'div' });

    expect(wrapper.type()).toBe('div');
  });

  it('should pass through other props', () => {
    const { wrapper } = setup({ tabIndex: 1, id: 'test' });

    expect(wrapper.prop('tabIndex')).toEqual(1);
    expect(wrapper.prop('id')).toEqual('test');
  });
});
