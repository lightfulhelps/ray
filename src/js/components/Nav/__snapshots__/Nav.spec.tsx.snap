// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should render 1`] = `
<ul
  className="nav"
/>
`;

exports[`<Nav /> should render with a wrapper 1`] = `
<div
  className="overflow-hidden"
>
  <ul
    className="nav"
  />
</div>
`;
